app/views/play/CampaignView.js

```diff
@@ -921,12 +925,31 @@ module.exports = (CampaignView = (function () {
         const campaignSlug = window.location.pathname.split('/')[2]
         return this.promptForSubscription(campaignSlug, 'premium campaign visited')
       }
+
+      if (
+        !me.get('email') &&
+        (storage.load('prompted-for-signup') || storage.load('prompted-for-subscription'))
+      ) {
+        if (!storage.load('roblox-modal-shown')) {
```

Comment:
we shouldn't be using strings `roblox-modal-shown` like this and reuse it in other function, can you move it to a constant?